docs/common/craft-parts/reference/plugins/dotnet_v2_plugin.rst

```diff
@@ -0,0 +1,231 @@
+.. _craft_parts_dotnet_v2_plugin:
+
+.NET plugin (v2)
+================
+
+The ``dotnet`` plugin builds .NET projects using the ``dotnet`` tool.
+
+.. note::
+    This plugin supersedes :ref:`.NET plugin <craft_parts_dotnet_plugin>`.
+
+Keywords
+--------
+
+In addition to the common :ref:`plugin <part-properties-plugin>` and
+:ref:`sources <part-properties-sources>` keywords, this plugin provides the following
+plugin-specific keywords:
+
+.. _global_flags:
+
+Global Flags
+~~~~~~~~~~~~
+
+dotnet-configuration
+^^^^^^^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``"Release"``
+
+The .NET build configuration to use. Possible values are ``"Debug"`` and ``"Release"``.
+
+dotnet-project
+^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``None``
+
+Path to the solution or project file to build relative to the root of the snap source.
+If a project or solution file is not specified, MSBuild will search t he root of the snap
+source for a file that has an extension that ends in either ``proj`` or ``sln``.
+
+dotnet-properties
+^^^^^^^^^^^^^^^^^
+**Type:** dict of strings to strings
+
+**Default:** Empty
+
+A list of MSBuild properties to be appended to the restore, build, and publish commands
+in the format of ``-p:<Key>=<Value>``.
+
+.. _dotnet_self_contained:
+
+dotnet-self-contained
+^^^^^^^^^^^^^^^^^^^^^
+**Type:** boolean
+
+**Default:** ``False``
+
+Create a self-contained .NET application. The Runtime Identifier (RID) will be automatically
+set based on the ``$CRAFT_BUILD_FOR`` variable for a given build, such that:
+
++------------------------------+------------------------+
+| ``$CRAFT_BUILD_FOR`` value   | .NET RID               |
++==============================+========================+
+| ``amd64``                    | ``linux-x64``          |
++------------------------------+------------------------+
+| ``arm64``                    | ``linux-arm64``        |
++------------------------------+------------------------+
+
+dotnet-verbosity
+^^^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``"normal"``
+
+Sets the MSBuild log output verbosity for the build. Possible values are:
+``q[uiet]``, ``m[inimal]``, ``n[ormal]``, ``d[etailed]``, and ``diag[nostic]``.
+
+.. _dotnet_version:
+
+dotnet-version
+^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``None``
+
+Sets the .NET version to build the project with. By setting this property, the plugin
```

Comment:
```suggestion
Sets the .NET version to build the project with. By setting this key, the plugin
```

When we mean the named YAML key, let's use "key". This is a separate concern from the MSBuild properties.